### PR TITLE
[IL-154]: create header with hero

### PR DESCRIPTION
**Pull request by iakivpekarskyi**
Finished header with adaptive feature (only mobile is available). Added hero image in the hero component but didn't create the new hero branch. I have one  issue with hero section - there is top margin and i don't know where it is coming from.

### CODE REVIEW

File: src/app/layout.tsx [hunk 1]
```diff
@@ -17,7 +19,10 @@ interface RootLayoutProps {
 export default function RootLayout({ children }: RootLayoutProps) {
   return (
     <html lang="en">
-      <body className={inter.className}>{children}</body>
+      <body className={inter.className}>
+        <Header />
+        <Hero/>
```
Comment (Wdyffs) [inline src/app/layout.tsx:24]: I can not see <Hero /> component in components folder

**Review by Wdyffs — COMMENTED** [review]
Rename pull request to [IL-154]: create header with hero

File: src/components/header.tsx
[the remark this answers is not in the document]
Comment (iakivpekarskyi) [reply src/components/header.tsx:1]: I tried everything except resetting the vs code settings but i still get auto-formatting when i press save and single quotes turn into double quotes.

**Comment by iakivpekarskyi** [comment]
Still have two issues:

1. Major one - conditional rendering of the hero component. I am not sure if the layout should be rendered on the clients side. If i use the client side rendering it is not allowed to export "metadata" cause it is marked "use client".
Screenshot 2023-11-04 at 23 45 50
Screenshot 2023-11-04 at 23 49 44

2. Minor issue - single quotation when using tailwind in VS code. Some auto-formatting overrides prettier and settings.json settings.

**Comment by Wdyffs** [comment]
> Still have two issues:
> 
> 1. Major one - conditional rendering of the hero component. I am not sure if the layout should be rendered on the clients side. If i use the client side rendering it is not allowed to export "metadata" cause it is marked "use client".
>    Screenshot 2023-11-04 at 23 45 50
>    Screenshot 2023-11-04 at 23 49 44
> 2. Minor issue - single quotation when using tailwind in VS code. Some auto-formatting overrides prettier and settings.json settings.



> Still have two issues:
> 
> 1. Major one - conditional rendering of the hero component. I am not sure if the layout should be rendered on the clients side. If i use the client side rendering it is not allowed to export "metadata" cause it is marked "use client".
>    Screenshot 2023-11-04 at 23 45 50
>    Screenshot 2023-11-04 at 23 49 44
> 2. Minor issue - single quotation when using tailwind in VS code. Some auto-formatting overrides prettier and settings.json settings.

1. You should not prefix RootLayous as a client component. Only page where hero exists is a products page ( it is, src/app/page.tsx). Hero is a server component, because it don't work with user interaction. Thus you should place Hero component in the src/app/_components folder

File: src/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import logo from '../../public/logo.svg'
+
+import Image from 'next/image'
+import Link from 'next/link'
+import CartButton from './CartButton'
+
+export default function Header() {
+  return (
+    <section
```
Comment (Wdyffs) [inline src/components/Header.tsx:9]: use tag header rather than section. Reason - CEO

**Review by Wdyffs — APPROVED** [review]
Review approved!

**Pull request merged**